### ISSUE TITLE
Silence error output

### ISSUE THREAD

**Issue by jrwebdev** [storybooks/storybook#1889]
When using `start-storybook`, we're seeing the whole compilation object output on error:

```
 @ multi ./node_modules/@storybook/react/dist/server/config/polyfills.js ./node_modules/@storybook/react/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./config/storybook/config.js
Stats {
  compilation:
   Compilation {
     _plugins:
      { 'child-compiler': [Object],
        record: [Object],
        'after-hash': [Object],
        'should-generate-chunk-assets': [Object],
        'need-additional-pass': [Object],
        'additional-chunk-assets': [Object],
        'build-module': [Object],
        'failed-module': [Object],
        'succeed-module': [Object],
        seal: [Object],
        optimize: [Object],
        'optimize-modules-basic': [Object],
        'optimize-modules': [Object],
        'optimize-modules-advanced': [Object],
        'optimize-chunks-basic': [Object],
        'optimize-chunks': [Object],
```

The actual output spans hundreds of lines. The compilation error is output correctly before all of this output, and it does not occur when using `build-storybook`. Having a look at the source, I've tracked it down to this line:

https://github.com/storybooks/storybook/blob/master/app/react/src/server/index.js#L162

Where it just logs to the console anything it has caught.

I'm not sure why we're getting the output shown above, but maybe it'd be good to have an option to turn off any caught errors being output to the console? Happy to submit a PR, just unsure what the solution should be.

**Comment by danielduan** [comment]
It sounds like you're probably using a custom webpack config but it's getting an error during validation.

Can you show us the config you're using?

**Comment by astrotim** [comment]
I have added `stats: 'none'` to my webpack.config.js which has suppressed the output on compilation error.

```javascript
module.exports = {
  stats: 'none',
  module: {
    ... etc etc
  }
}; 
```

**Comment by jrwebdev** [comment]
Thanks for responding. I've had a look into this a bit more and experimented with a number of dev server options, even the following causes the stats object to be output in its entirety:

```
{ noInfo: true,
  publicPath: '/',
  watchOptions: {},
  stats: 'none',
  quiet: true }
```

Looking through the code some more, these options are [passed to the dev middleware](https://github.com/storybooks/storybook/blob/d39713519b08efd6baef89a902fbc5721ab32b56/app/react/src/server/middleware.js#L39), which [rejects the promise if there are errors](https://github.com/storybooks/storybook/blob/master/app/react/src/server/middleware.js#L63). This is then just logged to the console as per the link in my first post.

Seems like this shouldn't be the behaviour, given the errors etc. are output to the console correctly prior to the stats object being logged.

@danielduan I can't really post the custom config in here as ours is very modular to support a number of projects, but I'm guessing the only thing that's relevant looking at the code above is the `devServer` config?

**Comment by danielduan** [comment]
This should be fixed by #2199 and published by now.
